src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -543,322 +790,197 @@ def __init__(self, config: BartConfig, embed_tokens, **kwargs):
-            input_ids = input_ids[:, -1:]
-            positions = positions[:, -1:]
+        if inputs_embeds is None:
+            inputs_embeds = self.embed_tokens(inputs["input_ids"])
+        else:
+            inputs_embeds = inputs["inputs_embeds"]
 
-        x = self.embed_tokens(input_ids) * self.embed_scale
-        if self.do_blenderbot_90_layernorm:
-            x = self.layernorm_embedding(x) + positions
+        hidden_states = inputs_embeds * self.embed_scale
+
+        # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
+        combined_attention_mask = None
+        if input_shape[-1] > 1:
+            combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
+
+        if inputs["attention_mask"] is None and inputs["input_ids"] is not None and input_shape[-1] > 1:
+            attention_mask = tf.cast(
+                tf.math.not_equal(inputs["input_ids"], self.config.pad_token_id), inputs["input_ids"].dtype
+            )
         else:
-            x = self.layernorm_embedding(x + positions)
-        x = self.dropout(x, training=training)
+            attention_mask = tf.ones(input_shape, dtype=tf.int32)
+
+        if attention_mask is not None and combined_attention_mask is not None:
+            # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
+            combined_attention_mask = combined_attention_mask + _expand_mask(
+                attention_mask, past_key_values_length=past_key_values_length
+            )
+
+        encoder_hidden_states = inputs["encoder_hidden_states"]
+        if encoder_hidden_states is not None and inputs["encoder_attention_mask"] is not None:
+            # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
+            encoder_attention_mask = _expand_mask(inputs["encoder_attention_mask"], tgt_len=input_shape[-1])
 
-        # Convert to Bart output format: (BS, seq_len, model_dim) ->  (seq_len, BS, model_dim)
-        x = tf.transpose(x, perm=(1, 0, 2))
-        assert len(shape_list(encoder_hidden_states)) == 3, "encoder_hidden_states must be a 3D tensor"
-        encoder_hidden_states = tf.transpose(encoder_hidden_states, perm=(1, 0, 2))
+        if self.do_blenderbot_90_layernorm:
+            hidden_states = self.layernorm_embedding(hidden_states) + positions
+        else:
+            hidden_states = self.layernorm_embedding(hidden_states + positions)
+        hidden_states = self.dropout(hidden_states, training=inputs["training"])
 
         # decoder layers
         all_hidden_states = ()
         all_self_attns = ()
-        next_decoder_cache = []
+        present_key_values = ()
         for idx, decoder_layer in enumerate(self.layers):
-            # add LayerDrop (see https://arxiv.org/abs/1909.11556 for description)
-            if output_hidden_states:
-                all_hidden_states += (x,)
+            # add LayerDrop (see https://arhidden_statesiv.org/abs/1909.11556 for description)
```

Comment:
lol the hidden state of arxiv
```suggestion
            # add LayerDrop (see https://arxiv.org/abs/1909.11556 for description)
```

Reply:
haha my search and replace gone wrong

Reply:
I need better regexes

Reply:
If you use vim I like to do `:%s/xxx/yyy/gc`, the `c` at the end is slower but helps! You probably know it already though, I'm not the king of vim, you are :)